inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -24,8 +25,12 @@ def __init__(self, db_file=UDM_DB_FILE) -> None:
         """
         self._db_file = db_file
         if self._db_file != ":memory:" and not os.path.exists(self._db_file):
-            raise DispatcherException(f"Error: Database file {self._db_file} does not exist.")
-
+            # Create database file if not exist
+            logger.info(f"Database file doesn't exist. Creating the file.")
+            file_descriptor = os.open(self._db_file, os.O_CREAT)
+            # Set permission of the file (rw for owner and group)
+            os.chmod(self._db_file, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP)
+            os.close(file_descriptor)
         try:
             with sqlite3.connect(self._db_file) as conn:
                 self._conn = conn                    
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Ensure database connections are always properly closed to prevent resource leaks.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        self._conn = sqlite3.connect(self._db_file)
        try:
            # Perform database operations
        finally:
            self._conn.close()
```

Reply:
`with` already ensures the connection is closed, and the AI suggestion is no longer catching exceptions from the connect call